src/CloudFoundry.VisualStudio/Controls/RoutesUserControl.xaml.cs

```diff
@@ -28,14 +28,15 @@ public RoutesUserControl()
 
         private void btnAddHost_Click(object sender, RoutedEventArgs e)
         {
-            if (string.IsNullOrWhiteSpace(tbName.Text) == false)
-            {
-                var dataContext = (this.DataContext as PublishProfileEditorResources);
 
-                if (dataContext == null)
-                {
-                    throw new InvalidOperationException("DataContext is not a valid PublishProfileEditorResources");
-                }
+            var dataContext = (this.DataContext as PublishProfileEditorResources);
+
+            if (dataContext == null)
+            {
+                throw new InvalidOperationException("DataContext is not a valid PublishProfileEditorResources");
+            }
+            if (dataContext.ValidateRoute(tbName.Text))
```

Comment:
the ValidateRoutes() method should validate routes within the datacontext, and it shouldn't return any value.